### PR TITLE
feat(dyn-sampling): add new bias toggle to project details for prioritise by tx name [TET-717]

### PR DESCRIPTION
**Pull request by andriisoldatenko**
This PR adds new toggle for project details for prioritise by tx name

Tested locally:

image

### CODE REVIEW

File: src/sentry/dynamic_sampling/rules/utils.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ class RuleType(Enum):
     BOOST_LATEST_RELEASES_RULE = "boostLatestRelease"
     IGNORE_HEALTH_CHECKS_RULE = "ignoreHealthChecks"
     BOOST_KEY_TRANSACTIONS_RULE = "boostKeyTransactions"
+    PRIORITISE_BY_TX_NAME_RULE = "prioritiseByTxName"
```
Comment (priscilawebdev) [inline src/sentry/dynamic_sampling/rules/utils.py:34]: shall we make this plural to be consistent?

```suggestion
    PRIORITIZE_BY_TRANSACTION_NAMES_RULE = "prioritizeByTransactionNames"
```

Comment (andriisoldatenko) [reply src/sentry/dynamic_sampling/rules/utils.py:34]: yeah sure we can rename, I had a similar name but then decided to make it shorter.

File: tests/sentry/api/endpoints/test_project_details.py [hunk 2]
```diff
@@ -1325,6 +1325,7 @@ def test_get_dynamic_sampling_biases_manually_set_biases(self):
                 },
                 {"id": "ignoreHealthChecks", "active": True},
                 {"id": "boostKeyTransactions", "active": True},
+                {"id": "prioritiseByTxName", "active": False},
```
Comment (RaduW) [inline tests/sentry/api/endpoints/test_project_details.py:1328]:
```suggestion
                {"id": "boostLowVolumeTransactions", "active": False},
```

File: tests/sentry/api/endpoints/test_project_details.py [hunk 3]
```diff
@@ -1468,6 +1470,7 @@ def test_put_new_dynamic_sampling_rules_with_correct_flags(self):
             },
             {"id": "ignoreHealthChecks", "active": False},
             {"id": "boostKeyTransactions", "active": False},
+            {"id": "prioritiseByTxName", "active": False},
```
Comment (iambriccardo) [inline tests/sentry/api/endpoints/test_project_details.py:1473]: Maybe you missed the renaming here.

File: tests/sentry/api/endpoints/test_project_details.py [hunk 4]
```diff
@@ -1351,6 +1352,7 @@ def test_get_dynamic_sampling_biases_with_previously_assigned_biases(self):
                 },
                 {"id": "ignoreHealthChecks", "active": True},
                 {"id": "boostKeyTransactions", "active": True},
+                {"id": "prioritiseByTxName", "active": False},
```
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply tests/sentry/api/endpoints/test_project_details.py:1355]: good catch, let me fix it.

[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply tests/sentry/api/endpoints/test_project_details.py]: @iambriccardo fixed in b064f9c453

File: tests/sentry/api/endpoints/test_project_details.py (hunk 3 above)
Comment (andriisoldatenko) [reply tests/sentry/api/endpoints/test_project_details.py]: fixed in b064f9c453

**Review by iambriccardo — APPROVED** [review]

**Comment by iambriccardo** [comment]
We would also need a separate PR to update the logging system.

**Review by priscilawebdev — APPROVED** [review]
Checked with Andrii locally and everything looks good

**Pull request merged**